### PR TITLE
 DOC: Removed line about %S parses all the way up to nanosecond even if no

### LINKED ISSUE #49231: DOC: remove statement that %S parses all the way up to nanosecond even if no decimal place present
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.to_datetime.html

### Documentation problem

The sentence

> "%S" without "%f" will capture all the way up to nanoseconds if present as decimal places, and will also handle the case where the number of seconds is an integer.

isn't generally true - it's only true for the ISO8601 fastpath, and that's because of a [bug](https://github.com/pandas-dev/pandas/issues/12649) (which I'm looking at addressing)

Demonstration of the issue:
```python
>>> import pandas as pd
>>> pd.to_datetime('2000-01-01 00:00:00.123456', format='%Y-%m-%d %H:%M:%S')  # iso8601 fastpath
Timestamp('2000-01-01 00:00:00.123456')
>>> pd.to_datetime('2000|01|01 00:00:00.123456', format='%Y|%m|%d %H:%M:%S')  # no fastpath
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/megorelli/.venv/lib/python3.9/site-packages/pandas/core/tools/datetimes.py", line 1102, in to_datetime
    result = convert_listlike(np.array([arg]), format)[0]
  File "/home/megorelli/.venv/lib/python3.9/site-packages/pandas/core/tools/datetimes.py", line 430, in _convert_listlike_datetimes
    res = _to_datetime_with_format(
  File "/home/megorelli/.venv/lib/python3.9/site-packages/pandas/core/tools/datetimes.py", line 538, in _to_datetime_with_format
    res = _array_strptime_with_fallback(
  File "/home/megorelli/.venv/lib/python3.9/site-packages/pandas/core/tools/datetimes.py", line 473, in _array_strptime_with_fallback
    result, timezones = array_strptime(arg, fmt, exact=exact, errors=errors)
  File "pandas/_libs/tslibs/strptime.pyx", line 156, in pandas._libs.tslibs.strptime.array_strptime
ValueError: unconverted data remains: .123456
```

### Suggested fix for documentation

1. remove this from the `format : str, default None` section

> "%S" without "%f" will capture all the way up to nanoseconds if present as decimal places, and will also handle the case where the number of seconds is an integer.

2. remove this from the "Differences with strptime behavior" section

> "%S" without "%f" will capture all the way up to nanoseconds if present as decimal places.

### PR DESCRIPTION
**Pull request by sarvaSanjay**
…decimal place present from docstring

- [ ] closes #49231
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -969,9 +965,6 @@ def to_datetime(
     ...                format='%Y-%m-%d %H:%M:%S.%f')
     Timestamp('2018-10-26 12:00:00.000000001')
 
-    :const:`"%S"` without :const:`"%f"` will capture all the way
-    up to nanoseconds if present as decimal places.
-
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:974]: you also need to remove the example code that follows

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -787,10 +787,6 @@ def to_datetime(
         - :const:`"%f"` will parse all the way
           up to nanoseconds;
 
-        - :const:`"%S"` without :const:`"%f"` will capture all the way
-          up to nanoseconds if present as decimal places, and will also handle
-          the case where the number of seconds is an integer.
-
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:793]: Now, the preceding paragraph says "though note the following difference", but then there's only one difference.

Perhaps we can change it to something like
```
though note that "%f" will parse all the way up to nanoseconds
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by MarcoGorelli** [comment]
Thanks for starting this

Could you also link to the issue in the issue description please?

File: pandas/core/tools/datetimes.py (hunk 1 above)
Comment (sarvaSanjay) [reply pandas/core/tools/datetimes.py:974]: Thank you for bringing this to my notice, I have deleted the example code.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Could you fix up the failing CI check, and also fill in the template in the issue description? If you put the issue number in there it'll be closed automatically when the PR is merged

**Comment by sarvaSanjay** [comment]
Hi I am a bit confused as to how I can add the issue number in the issue description Could you please clarify as to what exactly that means

**Comment by MarcoGorelli** [comment]
sure - where it says `closes #xxxx `, replace `xxxx` with the issue number of the issue which this pull request is closing, see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

regarding the CI failure, it should be enough to run the `black` formatter on the file you've modified, I think there's just some trailing whitespace

**Comment by sarvaSanjay** [comment]
Hi,
With the exception of two tests all tests hava passed. However, I do not understand why the two tests have not passed. I am attaching screenshots of the error messages. 
This is the error message for the AVS Benchmarks test
ASV Benchmarks
This is the error for the Data Manager test
data manager error
If you could please explain to me whats causing these errors I shall fix them ASAP

File: pandas/core/tools/datetimes.py [hunk 3]
```diff
@@ -782,15 +782,7 @@ def to_datetime(
...
-        note the following differences:
-
-        - :const:`"%f"` will parse all the way
-          up to nanoseconds;
-
-        - :const:`"%S"` without :const:`"%f"` will capture all the way
-          up to nanoseconds if present as decimal places, and will also handle
-          the case where the number of seconds is an integer.
-
+        note that "%f" will parse all the way up to nanoseconds.
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:785]: No need to remove the formatting
```suggestion
        note that :const:`"%f"` will parse all the way up to nanoseconds.
```

**Comment by MarcoGorelli** [comment]
the errors look unrelated (general tip - don't attach screenshots as they're not searchable, just copy-and-paste the text)

File: pandas/core/tools/datetimes.py (hunk 3 above)
Comment (sarvaSanjay) [reply pandas/core/tools/datetimes.py]: Thank you, I have re-added the formatting.

**Comment by sarvaSanjay** [comment]
> (general tip - don't attach screenshots as they're not searchable, just copy-and-paste the text)
Thank you I shall keep that in mind.
> the errors look unrelated
Ah ok. Well, is there any way we can remove the tests somehow or will that not be necessary?

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me pending green, thanks @sarvaSanjay !

regarding CI, I'm hoping that just having restarted it it'll pass

**Comment by sarvaSanjay** [comment]
Thanks a lot @MarcoGorelli ! This was my first contribution and it definitely was a lot of fun so this definitely won't be my last :) 
A GIF History of Leonardo DiCaprio Raising Glasses

**Comment by MarcoGorelli** [comment]
no need to backport, the original PR never made it in to 1.5.0

**Pull request merged**